packages/engine/src/datastore/table/pool/agent.rs

```diff
@@ -12,6 +12,8 @@ use crate::{
 };
 
 /// An ordered collection of similar [`AgentBatch`]es for each group within a simulation run.
+///
+/// All fields (except for messages) that agents' have are persisted in the Agent Pool.
```

Comment:
@litvand does the outbox ever get written to the batches? Or does it get split off at some point in flush_changes or something.

Reply:
The outbox is the state message pool, which is why messages aren't persisted in the agent pool. After messages are written to the outbox, they stay there in memory, but implicitly become the previous step's state message pool, i.e. the inbox.

Reply:
Gotcha